src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -198,21 +199,63 @@ public override IActorRef SystemActorOf<TActor>(string name = null)
         /// </summary>
         public override void Abort()
         {
-            Aborting = true;
+            Volatile.Write(ref Aborting, true);
             Terminate();
         }
 
         /// <summary>Starts this system</summary>
         public void Start()
+        {
+            using (var cts = new CancellationTokenSource(Settings.CreationTimeout))
+                StartAsync(cts.Token).GetAwaiter().GetResult();
+        }
+
+        /// <summary>Starts this system</summary>
+        public async Task StartAsync(CancellationToken cancellationToken = default)
         {
             try
             {
                 // Force TermInfoDriver to initialize in order to protect us from the issue seen in #2432
                 typeof(Console).GetProperty("BackgroundColor").GetValue(null); // HACK: Only needed for MONO
 
                 RegisterOnTermination(StopScheduler);
-                _provider.Init(this);
-                LoadExtensions();
+
+                //init provider
+                {
+                    _provider.Init(this);
+                    if (_provider is IInitializable init)
+                        await init.InitializeAsync(cancellationToken);
+                }
+
+                //load and init extensions
+                {
+                    var extensions = LoadExtensions();
+                    foreach (var init in extensions.OfType<IInitializable>())
+                        await init.InitializeAsync(cancellationToken);
```

Comment:
I think this is part of your problem - the old system had ordering in-place created as an effect of lazy loading. If `Cluster` depends on the `RARP` extension, `Cluster` would start `RARP` if it wasn't created already and would load the cached value.

Scanning the assembly and arbitrarily front-loading the extensions is non-deterministic and will result in the types of deadlocks you're seeing now.

I wouldn't front-load any of the extensions at all - lazy loading until they're needed on-demand is still the better approach.

Reply:
BTW, one simple fix here to see if it resolves the deadlock - don't `await` each individual plugin starting up. Compose them into a `WhenAll` and `await` on the group.

Reply:
Only Cluster and the RemoteActorRefProvider are supporting it,
yes the issue is WHO and WHEN the Cluster Extensions get cold and exit